### ISSUE TITLE
Unhandled promise rejection from twich-chat-client.connect while beeing offline

### ISSUE THREAD

**Issue by ruohki** [d-fischer/twitch#108]
## Bug Report
When trying to connect to twitch chat while beeing offline it leads to an unhandled promise rejection even if you use try/catch .catch

### Code
```js
const ChatClient = require('twitch-chat-client').default;
try {
  const chatClient = ChatClient.anonymous({
    readOnly: true,
    ssl:true,
    webSocket: true,
  });
  
  chatClient.onRegister(() => {
    console.log("Registerd")
  })
  

  chatClient.connect().catch(err => true);
} catch (ex) {
  
}
```

### Expected behavior
Handled promise rejection

### Actual Behavior
Unhandled promise rejection

### Environment

- Version: v10.16.3 & v12.16.1
- Node version: 
- Operating system: Win10 / Arch

**Comment by d-fischer** [comment]
The error is actually caught (via the promise-style `.catch()`. The error that is displayed is actually from the retry.

What needs to be done:
- Introduce a (configurable) limit to retries
- On the initial `.connect()` call, only reject the returned promise after all these retries failed
- Don't throw when a reconnect fails after being disconnected from a previously successful connection

**Comment by d-fischer** [comment]
This was finally fixed and released in the 4.1.0-pre.12 prerelease - please test it!
